src/MetadataManager/PathToHash.ts

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as vscode from 'vscode';
+import fs from 'fs';
+import * as crypto from 'crypto';
+
+/**
+ * A Singleton Object
+ * 
+ * PathToHash is a data structure that stores the path of the actual data 
+ * and the hash value generated by the contents of the actual data
+ * to synchronize the actual data and metadata
+ */
+
+class SyncMetadata{
+  static async validateMetadata(flattenMap: any) {
+
+    /**  1. Create metadata if pathToHash exists but does not have actual metadata files,
+     *      If pathToHash exists and there is a actual metadata file, but there is no path inside, create
+     *      a path and data insde
+     */
+    await this.createMetadata(flattenMap);
+
+    // 2. Replace is_deleted with true for all metadata not in pathToHash
+    await this.deleteMetadata(flattenMap);
+  }
+
+  static async createMetadata(flattenMap: any) {
+    for (let path in flattenMap) {
+      const hash = flattenMap[path];
+
+      // Metadata.createDefault(path-> uri, hash);
+
+    }
+  }
+
+  static async deleteMetadata(flattenMap: any) {
+    if (vscode.workspace.workspaceFolders === undefined) {
+      return;
+    }
+    const baseUri =
+        vscode.Uri.joinPath(vscode.workspace.workspaceFolders[0].uri, '.meta/hash_objects');
+    const files = await vscode.workspace.fs.readDirectory(baseUri);
+    for (const file of files) {
+      const hashFolderUri = vscode.Uri.joinPath(baseUri, file[0]);
+      const hashList = await vscode.workspace.fs.readDirectory(hashFolderUri);
+      for (const hashFile of hashList) {
+        const hashUri = vscode.Uri.joinPath(hashFolderUri, hashFile[0]);
+        let metaObj =
+            JSON.parse(Buffer.from(await vscode.workspace.fs.readFile(hashUri)).toString());
+        const hash = file[0] + hashFile[0].split('.')[0];
+        for (const path in metaObj) {
+          const uri = vscode.Uri.file(path);
+          // Metadata.disable(uri, hash);
+        }
+      }
+    }
+  }
+}
+
+export class PathToHash {
+  private static _instance: PathToHash;
+  private _map: any;
+
+  private constructor() {}
+
+  public static async getInstance() {
+    if (!this._instance) {
+      this._instance = new PathToHash();
+      this._instance._map = await this._instance.init();
+      await SyncMetadata.validateMetadata(await this._instance.getFlatMap());
```

Comment:
How about including this line into init() function, and only calls init function here?

Reply:
if I put this in init(), only getFlatMap will change a little. I'll take it into consideration